### DISCUSSION TITLE
CryptoStorm how to use it?

### DISCUSSION THREAD

**Discussion by d4g79** [haugene/docker-transmission-openvpn#2349] [General]
---
version: "3.4"
services:
  transmission-openvpn:
    image: haugene/transmission-openvpn:latest
    container_name: transmission-openvpn
    restart: always
    cap_add:
      - NET_ADMIN
    volumes:
      - /srv/
      - /srv/
    environment:
      - OPENVPN_PROVIDER=CRYPTOSTORM
      - OPENVPN_USERNAME= token that was given to me by cryptostorm
      - OPENVPN_PASSWORD=text
      - LOCAL_NETWORK=
      - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
      - PUID=1000
      - PGID=100
      - TZ=Australia/Sydney
      - TRANSMISSION_WEB_UI=combustion
      - TRANSMISSION_BLOCKLIST_ENABLED=true
      - TRANSMISSION_BLOCKLIST_URL=https://github.com/Naunter/BT_BlockLists/raw/master/bt_blocklists.gz
      - TRANSMISSION_DHT_ENABLED=false
      - TRANSMISSION_DOWNLOAD_DIR=
      - TRANSMISSION_INCOMPLETE_DIR= 
      - TRANSMISSION_PEER_LIMIT_GLOBAL=1200
      - TRANSMISSION_PEER_LIMIT_PER_TORRENT=240
      - TRANSMISSION_PEX_ENABLED=false
      - TRANSMISSION_SCRIPT_TORRENT_DONE_ENABLED=true
      - TRANSMISSION_SCRIPT_TORRENT_DONE_FILENAME=
      - TRANSMISSION_UMASK=0
      - TRANSMISSION_WATCH_DIR=/torrents/watch
    logging:
        driver: json-file
        options:
            max-size: 10m
    ports:
      - 9091:9091
      - 8888:8888
    dns:
      - 8.8.8.8
      - 8.8.4.4

The above config is what i sued but keep getting the below error.

Options error: You must define TUN/TAP device (--dev)

then it just trys again and the same error comes up.

Starting container with revision: a95f586492b2b7e8ff2cc8ac7f91bc43c0cccc40

Creating TUN device /dev/net/tun

Using OpenVPN provider: CRYPTOSTORM

Running with VPN_CONFIG_SOURCE auto

No bundled config script found for CRYPTOSTORM. Defaulting to external config

Downloading configs from https://github.com/haugene/vpn-configs-contrib/archive/main.zip into /tmp/tmp.fCjCKt8bNp

Extracting configs to /tmp/tmp.4BPK66O803

Found configs for CRYPTOSTORM in /tmp/tmp.4BPK66O803/vpn-configs-contrib-main/openvpn/cryptostorm, will replace current content in /etc/openvpn/cryptostorm

Cleanup: deleting /tmp/tmp.fCjCKt8bNp and /tmp/tmp.4BPK66O803

No VPN configuration provided. Using default.

Modifying /etc/openvpn/cryptostorm/default.ovpn for best behaviour in this container

Modification: Point auth-user-pass option to the username/password file

sed: can't read /etc/openvpn/cryptostorm/default.ovpn: No such file or directory

Modification: Change ca certificate path

sed: can't read /etc/openvpn/cryptostorm/default.ovpn: No such file or directory

Modification: Change ping options

sed: can't read /etc/openvpn/cryptostorm/default.ovpn: No such file or directory

sed: can't read /etc/openvpn/cryptostorm/default.ovpn: No such file or directory

sed: can't read /etc/openvpn/cryptostorm/default.ovpn: No such file or directory

sed: can't read /etc/openvpn/cryptostorm/default.ovpn: No such file or directory

Modification: Update/set resolv-retry to 15 seconds

Modification: Change tls-crypt keyfile path

Modification: Set output verbosity to 3

Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop

Setting OpenVPN credentials...

adding route to local network 192.168.2.0/24 via 172.25.0.1 dev eth0

Sat Sep 17 16:07:11 2022 disabling NCP mode (--ncp-disable) because not in P2MP client or server mode

Options error: You must define TUN/TAP device (--dev)

Use --help for more information.

thanx

**Comment by edgd1er** [comment]
the documentation is your friend:
* vpn configuration: cryptostorm has no default configuration, you have to define [one](https://haugene.github.io/docker-transmission-openvpn/supported-providers/#external_providers). here is the list: https://github.com/haugene/vpn-configs-contrib/tree/main/openvpn/cryptostorm
* either use CREATE_TUN_DEVICE=true or define /dev/tun device in your compose: https://haugene.github.io/docker-transmission-openvpn/config-options/#network_configuration_options
* LOCAL_NETWORK if empty, you may have difficulty to reach transmission: https://haugene.github.io/docker-transmission-openvpn/vpn-networking/
